### PR TITLE
tech: Create Default ProductCard story

### PR DESCRIPTION
**Pull request by Youakeem**
## Describe your changes
Compose the default `ProductCard` component out of different building blocks

<div class='graphite__hidden'>
          <div>🎥 Video uploaded on Graphite:</div>
            <a href="https://app.graphite.dev/media/video/RV4Oucd6mLf2QBBMAjRa/d92a9534-6d2d-4fd3-8ff2-3e40ea334d1f.mov">
              
            </a>
          </div>
<video src="https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/RV4Oucd6mLf2QBBMAjRa/d92a9534-6d2d-4fd3-8ff2-3e40ea334d1f.mov">Screen Recording 2024-09-16 at 14.41.09.mov</video>

## Justify why they are needed
New `ProductCard` components
https://www.figma.com/design/5kmmDdh6StpXzbEfr7WevV/Hedvig-UI-Kit?node-id=16269-14623&t=KfZGhlcQELqgimtF-1

## Checklist before requesting a review

- [x] I have performed a self-review of my code

### CODE REVIEW

**Comment by Youakeem** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4700?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4700** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4700?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4699** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4699?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4698** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4698?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4697** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4697?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4696** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4696?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4690** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4690?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Youakeem and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/store/src/components/ProductCard/ProductCard.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+            <BasePillow fill={tokens.colors.amber300} />
+          </Card.Media>
+          <Card.Heading>
+            <Card.Title>Homeowner Insurance</Card.Title>
+            <Card.Subtitle>Bellmansgatan 19A</Card.Subtitle>
+          </Card.Heading>
+        </Card.Header>
+
+        <ProductCardDetails.Root>
+          <ProductCardDetails.Trigger>
```
Comment (alebedev) [inline apps/store/src/components/ProductCard/ProductCard.stories.tsx:54]: Trigger part could be simpler
```
<ProductCardDetails.Trigger collapsedLabel="Show details" expendedLabel="hideDetails">
```

This is both easier to read and you can now express that both properties are required

File: apps/store/src/components/ProductCard/ProductCardDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+import Collapsible from '@/components/Collapsible/Collapsible'
+
+type ContextValue = {
+  isOpen: boolean
+  toggle: () => void
+}
+
+const Context = createContext<ContextValue | null>(null)
+
+const useProductDetails = () => {
```
Comment (alebedev) [inline apps/store/src/components/ProductCard/ProductCardDetails.tsx:12]: You probably no longer need a context and a hook to consume it once `Trigger` becomes a single component

Comment (Youakeem) [reply apps/store/src/components/ProductCard/ProductCardDetails.tsx:12]: We would still need it to render the right text, there is no way to know the open state inside of `Trigger` right?

File: apps/store/src/components/ProductCard/ProductCard.stories.tsx (hunk 1 above)
Comment (alebedev) [reply apps/store/src/components/ProductCard/ProductCard.stories.tsx:54]: We can pass JSX to props just as easily - any prop that is `ReactNode` will accept JSX

Comment (alebedev) [reply apps/store/src/components/ProductCard/ProductCard.stories.tsx:54]: Thinking what the good cases for nesting could be
- Some child needs to be deeper in the tree and surrounded by other components
- Child is optional, there could be 0, 1 or N in a given parent

Comment (Youakeem) [reply apps/store/src/components/ProductCard/ProductCard.stories.tsx:54]: > We can pass JSX to props just as easily - any prop that is ReactNode will accept JSX

Yup, that's what I meant!

File: apps/store/src/components/ProductCard/ProductCardDetails.tsx (hunk 2 above)
Comment (Youakeem) [reply apps/store/src/components/ProductCard/ProductCardDetails.tsx:12]: Ah, yes, true!

**Review by guilhermespopolin — APPROVED** [review]

File: apps/store/src/components/ProductCard/ProductCardDetails.tsx (hunk 2 above)
Comment (Youakeem) [reply apps/store/src/components/ProductCard/ProductCardDetails.tsx:12]: Thinking about this, we need the state in the Root as well so we can open and close the Collapsible. The state is being set in Trigger but read in both Trigger and Root.

You think we can do this without context?

File: apps/store/src/components/ProductCard/ProductCard.stories.tsx (hunk 1 above)
Comment (Youakeem) [reply apps/store/src/components/ProductCard/ProductCard.stories.tsx:54]: I changed the implementation to a `children` render prop. I think it hits the right balance between flexibility and readability. Let me know what you think!

**Comment by Youakeem** [comment]
### Merge activity

* **Sep 19, 3:16 AM EDT**: @Youakeem started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4700).

**Pull request merged**